app/controllers/spree/admin/users_controller.rb

```diff
@@ -100,6 +81,34 @@ def collection
 
       private
 
+      def update_message
+        return update_api_key_message if update_api_key_message.present?
+
+        if new_email_unconfirmed?
+          Spree.t(:email_updated)
+        else
+          Spree.t(:account_updated)
+        end
+      end
+
+      def update_api_key_message
+        return Spree.t(:show_api_key_view_toggled) if @user.show_api_key_view_previously_changed?
+        return t('spree.api.key_cleared') if api_key_cleared?
+
+        t('spree.api.key_generated') if api_key_generated?
+      end
+
+      def api_key_cleared?
+        @user.spree_api_key_previously_changed?(from: @user.spree_api_key_previously_was, to: nil)
```

Comment:
I think that we can simplify this. The `from: @user.spree_api_key_previously_was` would always match.
```suggestion
        @user.spree_api_key_previously_changed?(to: nil)
```